packages/camp/package.py

```diff
@@ -183,17 +185,25 @@ def cmake_args(self):
                 flag = "-arch sm_{0}".format(cuda_arch[0])
                 options.append("-DCMAKE_CUDA_FLAGS:STRING={0}".format(flag))
 
-        options.append(self.define_from_variant("ENABLE_HIP", "rocm"))
-        if "+rocm" in spec:
-            options.append("-DHIP_ROOT_DIR={0}".format(spec["hip"].prefix))
+        if "+rocm" in spec and not "+sycl" in spec:
```

Comment:
@long58 If I read correctly, +rocm will be ignored when +sycl.
What is your opinion about setting this as a `conflict` at the variant level, instead of dealing with it at the configuration level like here?
The danger here is not to know what target is being built: Spack will think it’s building "+rocm +sycl", while it looks like you are in facts building "~rocm +sycl".

Reply:
Yeah, that would probably make more sense. I'm very new to spack, and it kind of confuses me 😬, so I didn't really know all the ins and outs of what I was doing... But yeah, basically, if sycl is enabled, we don't want HIP enabled, so I guess it would make the most sense to just prevent sycl and rocm from being enabled at the same time.

Reply:
Everyone finds Spack confusing in some respect....